### PR TITLE
Generate portable pdbs in coreclr

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @dougbu \ @davidfowl - we saw some build failures on one of the CI agents for netcoreapp1.0 tests `'Windows PDB writer is not available -- could not find Microsoft.DiaSymReader.Native.amd64.dll'` whilst they ran fine on net451.

This change modifies compilation to always generate portable pdbs on netcoreapp.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -27,9 +27,14 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
...
-        private readonly DebugInformationFormat _pdbFormat = SymbolsUtility.SupportsFullPdbGeneration() ?
-            DebugInformationFormat.Pdb :
+        private readonly DebugInformationFormat _pdbFormat =
+#if NET451
+            SymbolsUtility.SupportsFullPdbGeneration() ?
+                DebugInformationFormat.Pdb :
+                DebugInformationFormat.PortablePdb;
+#else
             DebugInformationFormat.PortablePdb;
+#endif
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Why two `#if`s for one issue? Seems simpler to have one bit of conditional compilation in `SymbolsUtility.SupportsFullPdbGeneration()` -- return `false` when `!NET451`.

**Comment by dougbu** [comment]
> `'Windows PDB writer is not available -- could not find Microsoft.DiaSymReader.Native.amd64.dll'`

Where does this message come from and why doesn't the `try` / `catch` in `SymbolsUtility` handle it?

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: We end up having to `#ifdef` the usings too. So it ends up about the same number of `#ifdef`s

**Comment by pranavkm** [comment]
I'm not entirely sure. SymbolsUtility says the SymWriter component is installed (and I verified it does), but it doesn't work on CoreCLR unless we add a reference to the DiaSymReader package. Since portable pdbs work all the time on CoreCLR, this works pretty well.

**Pull request merged**